Third-party-libraries.md

```diff
@@ -212,7 +215,43 @@ Note that we don't have an automatic way to detect outdated libraries in `depend
+
+**For Small, Un-Compiled Open-Source Dependencies:** For small open-source dependencies where the code is executed directly, we can review the source code manually.
+
+* [ ] **We have manually reviewed the source code for the dependency and confirmed that it is safe:**
+* [ ] **Dependency compilation is safe.** This is essential because if the source code of a dependency is compiled and the compiled form is downloaded and executed, we cannot inspect the compiled form to ensure it is safe. We need to confirm one of the following:
+  * [ ] **The dependency is not compiled at all.** For example, shell scripts.
+  * [ ] **The dependency is compiled, and we have confirmed that the inspected source code produces the pinned compiled version bit-for-bit:** For example, locally compile the inspected source code and check that the output matches what we install.
+* [ ] **Our reference to the dependency is pinned by a hash.** To ensure malicious code cannot be introduced after our review, we must pin a hash of the specific code we reviewed. This could be a commit hash, for example. Note that we must verify for ourselves that the hash matches the code we reviewed. A commit hash from GitHub satisfies this requirement. However, if we reviewed the code of a Python dependency on GitHub and then wanted to rely on the checksum generated by pip, we would need to verify that the code downloaded by pip exactly matches the code we reviewed on GitHub.
+
+**For Proprietary Dependencies:** Proprietary software is generally produced by companies in-house using procedures not available for our review and may be closed-source. We therefore have to evaluate their safety by the reputation of the company making it.
+
+* [ ] **Compatible license:**
+* [ ] **Reputable author:** We're trusting the company to develop its software securely, so we must be confident that we can trust the author to (a) not be malicious, (b) carefully implement security best-practices to secure their software, and (c) keep their development systems safe from malicious actors. Since we cannot evaluate the software or how it's developed directly, this reputation bar is very high.
+* [ ] **Actively maintained based on recent release history:**
+
+#### How to use these checklists
+
+Copy the markdown source of the appropriate checklist, and paste it into your review comment, along with a link to this page. Then verify that each required criterion is satisfied and tick the associated checkbox. After the colon (`:`) for each ticked box, replace the help text with a description of why the requirement is satisfied. The OpenSSF Scorecard project is helpful for checking many of these points. For example, here's Oppia's scorecard: https://scorecard.dev/viewer/?uri=github.com/oppia/oppia. Finally, note whether the dependency is approved, and explain why if you reject the dependency. Here's an example of how to use these checklists: https://github.com/oppia/oppia/pull/20362#discussion_r1626804621
```

Comment:
Oh, now I see what's going on with the colons haha.

Maybe provide help text for each colon, not just some?

Reply:
done